app/models/ext_management_system.rb

```diff
@@ -76,6 +76,12 @@ def self.api_allowed_attributes
   has_many :miq_events,             :as => :target, :dependent => :destroy
   has_many :cloud_subnets, :foreign_key => :ems_id, :dependent => :destroy
 
+  has_many :configuration_scripts_and_workflows,
```

Comment:
Just name it configuration_scripts, and then you don't need the :class_name bit.  Also, if there is another type in the future, you won't need to rename the method to say a_and_b_and_c

Reply:
Naming `configuration_scripts` will collide (and overridden) by [this](https://github.com/jameswnl/manageiq/blob/dd1215d37906f6fd7ee757cdd5510a019bfa7039/app/models/manageiq/providers/automation_manager.rb#L14-L14)

Reply:
This name is very uncomfortable, but I have no better suggestion to resolve the naming conflict.

Reply:
@jameswnl If the `configuration_scripts ` association is already defined in the sub-class `ManageIQ::Providers::AutomationManager` as you point out why is this `has_many` needed at all?  The Tower provider would just use the existing association.

Reply:
Does it need to reference that name-spaced class or is it just a side effect?  Can we define a `:class_name` property on the existing association of `::ConfigurationScripts`.

Reply:
We can define `:class_name  => '::ConfigurationScript'` to existing `has_many :configuration_scripts` in `ManageIQ::Providers::AutomationManager`.  And then `tower_ems.configuration_scripts` will return both.  But then we'll lose the convenience of "get me only the job_templates of this ems"

In my mind we are trying to fulfill the following query needs:
* get all `job_templates/workflow_templates` of `vmdb`
* get all `job_templates/workflow_templates` of  this `ems`
  * This new `has_many :configuration_scripts_and_workflows` is for this purpose. We can drop this if we don't need it
  * Also this is really for convenience,  without this, we can do  `ConfigurationScript.where(:ems => some_tower_ems)` to achieve the effect
  * @h-kataria what do you think?
* get all `job_templates` of  this `ems`
* get all `workflow_templates` of  this `ems`

Reply:
My understanding was we wanted to reference them together for everything we have discussed so far.  If someone needs to differentiate between the two they could filter on type or we could provide scopes.  But in my mind separating them is not an immediate requirement for our use-case.

Reply:
in that case I'll move the one in `ManageIQ::Providers::AutomationManager` here.